child/child.go

```diff
@@ -373,7 +373,7 @@ func (c *Child) kill(immediately bool) {
 	exited := false
 	process := c.cmd.Process
 
-	if c.cmd.ProcessState != nil {
+	if !c.running() || c.stopped {
```

Comment:
For the other case `c.stopped`, when would this value be set to `true` other than when the child is stopped by an internal call with `child.Stop()` and `child.StopImmediately()`? The changes no longer seem to capture the scenario where the process had been stopped externally or killed prematurely, and is that intentional?

It looks like there isn't a great way to get information about the status of a process without this data race or explicitly waiting for it to exit. Not that this needs to be done here or in this PR, but I'm curious what you think about the wrapper `go-cmd/cmd` https://github.com/go-cmd/cmd/blob/master/cmd.go#L252

Reply:
Was a little to focused on just the fix it seems. That redundant `!c.running()` check can go and those 2 other variables should move down closer to where they are used. I'll fix that.

Reply:
I added the c.stopped check there for clarity/self-documentation. It was in the old code (before the patch that added the race) and felt it helped document that the stopped case was handled. It will never actually fire as c.stopped is set with a lock that prevents the race here.

Thinking about it I could see that case that it actually confuses things rather than clarifies them as it could make you think that there was a chance for it to actually fire ... Ok, convinced myself that it hurts more than it helps. Going to remove it.

Note that I'm not really happy with the organization of the code in this file and plan on re-organizing it at some point post-librarification.

Reply:
If the process is killed/dies outside the code `process.Wait()` will return with the exit code, send it down and close the `exitCh` channel. This way if `<-exitCh` has data to read that means the process is no longer running.

Reply:
Oh perfect, thanks for explaining the new flow. The dedup looks good too!